system/assets/hypha-preview-image-file-field.js

```diff
@@ -0,0 +1,64 @@
+$(function() {
+	$('img[data-preview-for]').each(function () {
+		// get preview
+		var $preview = $(this);
+
+		// get input name
+		var inputName = $preview.data('preview-for');
+
+		// get input
+		var $input = $('input[name="' + inputName + '"][type="file"]');
+
+		// add button to remove the image file
+		var addRemoveImageBtn = function () {
+			$('<span class="hyphaRemoveButton" />')
+				.on('click', function (e) {
+					e.preventDefault();
+
+					// remove delete button
+					$(this).remove();
+
+					updateSrc(null);
+				})
+				.insertAfter($preview);
+		}
+
+		var toggle = function (hasSrc) {
+			if (hasSrc) {
+				$input.hide();
+				$('label[for="' + inputName + '"]').attr('for', inputName + '_disabled');
+				addRemoveImageBtn();
+			} else {
+				$input.show();
+				$('label[for="' + inputName + '_disabled"]').attr('for', inputName);
+
+				// clear file value
+				$input.attr('value', null);
+			}
+		};
+
+		var updateSrc = function (src) {
+			let hasSrc = !!src;
+			if (hasSrc) {
+				$preview.attr('src', src);
+			} else {
+				$preview.removeAttr('src');
+			}
+
+			// update shadow file value
+			$('input[data-file-shadow-value-for="' + inputName + '"]').attr('value', src);
+
+			toggle(hasSrc);
+		}
+
+		var init = function () {
+			$input.on('change', function () {
+				updateSrc(window.URL.createObjectURL(this.files[0]));
```

Comment:
Does this produce a meaningful URL, or a data url? If the latter, there's no real point in saving it into the shadow field in `updateSrc`, I think?

Reply:
True, i just need to put something there, so we can detect if the image remained the same, got replaced or removed.

Reply:
But you do not need to detect "replaced" using the hidden field, I think, since that is detected when a new file upload has happened in the file input. Removed does need to be detected, but that can just empty out the field?

If we add a library-browser type thing later (where you can select existing images), you can just put the already known public URL of these images in the input field as well.

Reply:
fixed